### ISSUE TITLE
Release of MMPROJ and GGUF format files?

### ISSUE THREAD

**Issue by GJSea** [BAAI-DCAI/Bunny#64]
Hi,

Bunny-Llama-3-8B-V is very impressive and is able to pick up far more details than anything that uses clip-vit-large-patch14-336!

As most of the community uses [LLAMA.CPP](https://github.com/ggerganov/llama.cpp) to develop on, I was wondering if you might be able to publish the GGUF and MMPROJ pairing files for you model? It would get significant adoption if in this format.

Thx!

**Comment by Isaachhh** [comment]
Hi! We released GGUF [here](https://huggingface.co/BoyaWu10/Bunny-Llama-3-8B-V-gguf).

**Comment by GJSea** [comment]
Hi,

Thanks so much for the model files!

I might have found a bug in LLAVA.CPP, though I’m not sure if it is a data error in the positions tensor.


  *   In clip.cpp -> clip_image_batch_encode, num_positions is calculated to be 730.
  *   ggml_nbytes(positions) allocates positions_data buffer for 2,916 bytes.
  *   However for (int i = 0; i < num_positions; i++) ==> 730 * sizeof(int) = 2,920 which over indexes the positions_data buffer size of 2916
  *   free(positions_data); causes a heap corruption.

If I hard allocate positions_data  to be 2,920 bytes, your model works wonderfully.


So I’m not sure:


  *   If num_positions should be without the +1
  *   If  for (int i = 0; i < num_positions; i++)   should be  for (int i = 0; i < ggml_nbytes(positions) / sizeof(int); i++)
  *   Or if there is a data error in ggml_graph_get_tensor(gf, "positions");

I’m happy to file a bug with llama.cpp if this isn’t a data error in the MMPROJ file.

Thanks!
Gavin

**Comment by Isaachhh** [comment]
Hi, Gavin,

TL;DR `num_positions` should be without the +1

Actually, Bunny-Llama-3-8B-V is built upon **SigLIP**, while `llama.cpp` only supports CLIP now. We use the `clip.cpp` to load SigLIP weights temporarily now but we are not fully sure whether it's reasonable. And CLIP has a `CLS` token while `SigLIP` not, so `num_positions` should be without the +1.

However, although there are an out-of-bounds circumstance, I don't meet the heap corruption and the output is the same whether with the +1 or not. Thanks for pointing out that.

**Comment by Isaachhh** [comment]
This problem has been solved by https://github.com/ggerganov/llama.cpp/commit/d11afd665241c1b3910ab5f040d0216403019d87

**Comment by Isaachhh** [comment]
Close the issue for now if there's no further discussions. Feel free to reopen it if there's any other questions.

**Comment by GJSea** [comment]
Thanks so much, really appreciate your help!
________________________________
